src/ui-test/tests/camelk_java_debug_test.ts

```diff
@@ -0,0 +1,198 @@
+				this.timeout(60000);
+				await moveToExplorerActivity();
+				await createIntegration(INTEGRATION_FILE);
+				await modifyCurrentFileToBeInvalid();
+				await startIntegration(INTEGRATION_LABEL);
+			});
+
+			it('Test Java Debugger Not Available On Invalid File', async function() {
+				this.timeout(20000)
+				const item = await getIntegration(INTEGRATION_LABEL);
+				const menu = await item.openContextMenu();
+				await promiseFail(VSBrowser.instance.driver.wait(() => menu.hasItem(START_DEBUG_LABEL), 5000));
+			});
+
+			after(async function() {
+				this.timeout(20000);
+				await removeIntegration(INTEGRATION_LABEL);
+				await prepareEmptyTestFolder(WORKSPACE_FOLDER);
+			});
+		})
+
+	});
+
+	async function prepareTempWorkspaceForTests(workspaceFolder: string) {
+		await new EditorView().closeAllEditors();
+		await prepareEmptyTestFolder(workspaceFolder);
+		await VSBrowser.instance.openResources(workspaceFolder);
+	}
+
+	async function camelKToolingIsEnabled() {
+		const marketplace = await Marketplace.open();
+		const item = await marketplace.findExtension(`@installed ${pjson.displayName}`);
+		return extensionIsActivated(item, extensionActivated);
+	}
+
+	async function moveToExplorerActivity() {
+		const ACTIVITY_LABEL = 'Explorer';
+
+		return moveToActivity(ACTIVITY_LABEL);
+	}
+
+	async function moveToActivity(viewIdentifier: string) {
+		const control = await new ActivityBar().getViewControl(viewIdentifier);
+		return VSBrowser.instance.driver.wait(async () => control?.openView());
+	}
+
+	async function getCamelKIntegrationSection() {
+		const section = await new SideBarView().getContent().getSection(uiTestConstants.extensionName) as CustomTreeSection;
+		await section.expand();
+		await VSBrowser.instance.driver.wait(async () => section.isExpanded());
+		return section;
+	}
+
+	async function getIntegration(integrationLabel: string) {
+		const section = await getCamelKIntegrationSection();
+		await VSBrowser.instance.driver.wait(async () => hasIntegration(integrationLabel));
+		return await section.findItem(integrationLabel) as ViewItem;
+	}
+
+	async function hasIntegration(integrationLabel: string) {
+		const section = await getCamelKIntegrationSection();
+		return await sectionHasItem(section, integrationLabel);
+	}
+
+	async function createIntegration(integrationFile: string) {
+		const workbench = new Workbench();
+		await workbench.executeCommand(uiTestConstants.createNewIntegrationFile);
+		const languageInput = await InputBox.create();
+		await languageInput.selectQuickPick('Java');
+		const WORKSPACE_FOLDERInput = await InputBox.create();
+		await WORKSPACE_FOLDERInput.selectQuickPick(0);
+		const nameInput = await InputBox.create();
+		await nameInput.setText(integrationFile);
+		await nameInput.confirm();
+
+		const editorView = new EditorView();
+		await VSBrowser.instance.driver.wait(async() => {
+			try {
+				return await editorView.openEditor(integrationFile + '.java') !== undefined;
```

Comment:
Shouldn't the editor be opened when creating a new integration? Shouldn't we checked that the editor is opened instead of trying to open it?

Reply:
There is interest in UI Tests which are more end to end test to be closer to expected behavior from users too.
it won't change the potential flakyness.
it will fail this test in case the command is no more working that's right.
it slightly complexifies the codebase to add this code.

Both ways have there advantages and drawbacks. I tend to privilege to be closer to real user scenario for UI tests but that's tfine to privilege decoupling.